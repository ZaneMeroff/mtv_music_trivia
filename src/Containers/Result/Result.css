.result-contianer {
  align-items: center;
  display: flex;
  flex-direction: column;
  height: 88%;
  padding-top: 30px;
}

.result-text {
  font-size: 150%;
  margin: 1%;
  text-transform: uppercase;
}

.new-game-button {
  background-color: #fff200;
  border: none;
  font-size: 110%;
  height: 40px;
  border-radius: 20px;
  margin: 10% 0%;
  text-transform: uppercase;
  width: 200px;
}

.new-game-button:hover {
  background-color: #ff64de;
  color: #FFF;
  cursor: pointer;
}

#user-input {
  color: #fff200;
}

.missed-question-outer-container {
  align-items: center;
  display: flex;
  flex-direction: column;
  height: 70%;
  overflow-y: scroll;
  padding-top: 10px 10px 0px 10px;
}

.missed-question-container {
  background-color: #ff64de;
  border-radius: 20px;
  height: auto;
  margin: 3%;
  width: 70%;
}

.missed-question-text {
  font-size: 100%;
  padding: 1%;
  text-align: center;
}

.missed-question-data {
  color: #fff200;
}

@media only screen and (max-width: 490px) {

  .result-text {
    font-size: 90%;
  }

  .missed-question-text {
    font-size: 90%;
  }

  .new-game-button {
    font-size: 90%;
    height: 40px;
    border-radius: 20px;
  }

  .result-contianer {
    height: 85%;
  }

}
