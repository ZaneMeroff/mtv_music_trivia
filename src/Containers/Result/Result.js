import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import './Result.css';

export class Result extends Component {

  render() {

    let missedQuestions = this.props.wrong.map(q => {
      return (
        <div key={q.question} className='missed-question-container'>
          <p className='missed-question-text'>{q.question}</p>
          <p className='missed-question-text'>your answer: <span className='missed-question-data'>{q.your_answer}</span></p>
          <p className='missed-question-text'>correct answer: <span className='missed-question-data'>{q.correct_answer}</span></p>
        </div>
      )
    })

    return (
      <div className='result-contianer'>
        <p className='result-text'>way to go <span id='user-input'>{this.props.username}!</span></p>
        <p className='result-text'>score: <span id='user-input'>{this.props.right.length}0%</span></p>
        <Link to={'/'}><button className='new-game-button'>new game</button></Link>
        <div className='missed-question-outer-container'>
          {missedQuestions}
        </div>
      </div>
    )
  }
}

export const mapStateToProps = (state) => ({
  username: state.userName,
  right: state.correctQuestions,
  wrong: state.incorrectQuestions
});

export default connect(mapStateToProps)(Result);
