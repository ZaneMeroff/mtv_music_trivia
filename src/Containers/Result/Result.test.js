import React from 'react';
import { Result, mapStateToProps } from './Result';
import { shallow } from 'enzyme';

describe('Result', () => {
  let wrapper;
  let question;
  beforeEach(() => {
    question = [{question: 'Who da best band?'}]
    wrapper = shallow(<Result username={'Bob'} right={question} wrong={question} />)
  })
  it ('should render a Result component', () => {
    expect(wrapper).toMatchSnapshot()
  })
})

describe('mapStateToProps', () => {
  it('should return an object with a user, right questions array, and wrong questions array', () => {
    const question = [{question: 'Who da best band?'}]
    const mockState = {
      username: 'Bob',
      right: question,
      wrong: question
    };
    const expected = {
      username: 'Bob',
      right: question,
      wrong: question
    };
    const mappedProps = mapStateToProps(mockState);
    expect(mappedProps).toEqual(expected)
  });
});
