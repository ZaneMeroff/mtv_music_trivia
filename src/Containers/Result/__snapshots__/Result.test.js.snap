// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Result should render a Result component 1`] = `
<div
  className="result-contianer"
>
  <p
    className="result-text"
  >
    way to go 
    <span
      id="user-input"
    >
      Bob
      !
    </span>
  </p>
  <p
    className="result-text"
  >
    score: 
    <span
      id="user-input"
    >
      1
      0%
    </span>
  </p>
  <Link
    to="/"
  >
    <button
      className="new-game-button"
    >
      new game
    </button>
  </Link>
  <div
    className="missed-question-outer-container"
  >
    <div
      className="missed-question-container"
      key="Who da best band?"
    >
      <p
        className="missed-question-text"
      >
        Who da best band?
      </p>
      <p
        className="missed-question-text"
      >
        your answer: 
        <span
          className="missed-question-data"
        />
      </p>
      <p
        className="missed-question-text"
      >
        correct answer: 
        <span
          className="missed-question-data"
        />
      </p>
    </div>
  </div>
</div>
`;
